python/python/tests/test_async_client.py

```diff
@@ -797,6 +797,84 @@ async def test_hkeys(self, redis_client: TRedisClient):
         with pytest.raises(RequestError):
             await redis_client.hkeys(key2)
 
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_hrandfield(self, redis_client: TRedisClient):
+        key = get_random_string(10)
+        key2 = get_random_string(5)
+        field = get_random_string(5)
+        field2 = get_random_string(5)
+        field_value_map = {field: "value", field2: "value2"}
+
+        assert await redis_client.hset(key, field_value_map) == 2
+        assert await redis_client.hrandfield(key) in [field, field2]
+        assert await redis_client.hrandfield("non_existing_key") is None
+
+        assert await redis_client.set(key2, "value") == OK
+        with pytest.raises(RequestError):
+            await redis_client.hrandfield(key2)
+
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_hrandfield_count(self, redis_client: TRedisClient):
+        key = get_random_string(10)
+        key2 = get_random_string(5)
+        field = get_random_string(5)
+        field2 = get_random_string(5)
+        field_value_map = {field: "value", field2: "value2"}
+
+        assert await redis_client.hset(key, field_value_map) == 2
+        # Unique values are expected as count is positive
+        rand_fields = await redis_client.hrandfield_count(key, 4)
+        assert len(rand_fields) == 2
+        assert set(rand_fields) == {field, field2}
+
+        # Duplicate values are expected as count is negative
+        rand_fields = await redis_client.hrandfield_count(key, -4)
+        assert len(rand_fields) == 4
+        for rand_field in rand_fields:
+            assert rand_field in [field, field2]
```

Comment:
I think you can repeat here assert from line 830 - a set drops duplicates, but you checked the lenght

Reply:
This doesn't work because the command may return the same duplicated entry every time, or it may return different entries each time. In the first case the resulting set would be of length 1 but in the second case it would be of length 4.